src/utils.ts

```diff
@@ -49,18 +49,41 @@ ${summary}
 
 </details>`;
 
+const deletePreviousComments = async (issueNumber: number) => {
+  const octokit = getOctokit(getInput('github-token'));
+  const { data } = await octokit.issues.listComments({
+    ...context.repo,
+    per_page: 100,
+    issue_number: issueNumber
+  });
+
+  return Promise.all(
+    data
+      .filter(
+        (c) =>
+          c.user && c.body && c.user.login === 'github-actions[bot]' && c.body.indexOf('Total Coverage') !== -1
+      )
+      .map((c) => octokit.issues.deleteComment({ ...context.repo, comment_id: c.id }))
+  );
+};
```

Comment:
I think that this method is difficult to read. I think that can be more readable similar to this:

```js
    const { data: comments } = await octokit.issues.listComments({
        ...context.repo,
        per_page: 100,
        issue_number: issueNumber
    });
    return Promise.all(
        comments.filter(isPreviousTotalCoverageComment)
            .map(deleteComment(octokit)));
```

```js
const isPreviousTotalCoverageComment = comment => {
    comment.user &&
    comment.body &&
    comment.user.login === 'github-actions[bot]' &&
    comment.body.includes('Total Coverage');
};

const deleteComment = octokit => comment =>
    octokit.issues.deleteComment({ ...context.repo, comment_id: comment.id });
```

Reply:
I agree with you ! I'll update this :)